File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/data/type/LongValueType.ts
module api.data.type {

    export class LongValueType extends ValueType {

        constructor() {
            super("Long");
        }

        isValid(value: any): boolean {
            return api.util.NumberHelper.isWholeNumber(value);
        }

        isConvertible(value: string): boolean {
            if (api.util.StringHelper.isBlank(value)) {
                return false;
            }
            var convertedValue = Number(value);
            return this.isValid(convertedValue);
        }

        newValue(value: string): Value {
            if (!value) {
                return this.newNullValue();
            }

            if (!this.isConvertible(value)) {
                return this.newNullValue();
            }
            return new Value(this.convertFromString(value), this);
        }

        fromJsonValue(jsonValue: number): api.data.Value {
            return new Value(jsonValue, this);
        }

        private convertFromString(value: string): number {
            return Number(value);
        }

        valueToString(value: Value): string {
            return (<Number>value.getObject()).toString();
        }
    }
}

File: modules/core/core-api/src/main/java/com/enonic/xp/data/LongValueType.java
package com.enonic.xp.data;

final class LongValueType
    extends ValueType<Long>
{
    LongValueType()
    {
        super( "Long", JavaTypeConverters.LONG );
    }

    @Override
    public Value fromJsonValue( final Object object )
    {
        return ValueFactory.newLong( convertNullSafe( object ) );
    }
}
